addons/portal/views/portal_templates.xml

```diff
@@ -76,22 +76,22 @@
            t-nocache-_btn_class="_btn_class"
            t-nocache-_txt_class="_txt_class"
            t-nocache-_dropdown_menu_class="_dropdown_menu_class">
-            <t t-set="active_lang" t-value="list(filter(lambda lg : lg[0] == lang, languages))[0]"/>
-            <t t-set="language_selector_visible" t-value="len(languages) &gt; 1"/>
+            <t t-set="active_lang" t-value="frontend_languages[lang]"/>
+            <t t-set="language_selector_visible" t-value="len(frontend_languages) &gt; 1"/>
             <div t-attf-class="js_language_selector #{_div_classes} d-print-none" t-if="language_selector_visible">
                 <button t-attf-class="btn border-0 dropdown-toggle #{_btn_class or 'btn-sm btn-outline-secondary'}" type="button" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
                     <span t-if="not no_text"
                             t-attf-class="#{_txt_class}"
-                            t-esc="active_lang[2].split('/').pop()"/>
-                    <span t-elif="codes" class="align-middle" t-esc="active_lang[1].split('_').pop(0).upper()"/>
+                            t-esc="active_lang.name.split('/').pop()"/>
+                    <span t-elif="codes" class="align-middle" t-esc="active_lang.url_code.split('_').pop(0).upper()"/>
                 </button>
                 <div t-attf-class="dropdown-menu #{_dropdown_menu_class}" role="menu">
-                    <t t-foreach="languages" t-as="lg">
-                        <a class="dropdown-item" t-att-href="url_for(request.httprequest.path + '?' + keep_query(), lang_code=lg[0])"
+                    <t t-foreach="frontend_languages.values()" t-as="lg">
+                        <a class="dropdown-item" t-att-href="url_for(request.httprequest.path + '?' + keep_query(), lang_code=lg.code)"
                         t-attf-class="dropdown-item js_change_lang #{active_lang == lg and 'active'}"
-                        t-att-data-url_code="lg[1]" t-att-title="lg[2].split('/').pop()">
-                            <span t-if="not no_text" t-esc="lg[2].split('/').pop()" t-attf-class="#{_txt_class}"/>
-                            <span t-elif="codes" t-esc="lg[1].split('_').pop(0).upper()" t-attf-class="align-middle #{_txt_class}"/>
+                        t-att-data-url_code="lg.url_code" t-att-title="lg.name.split('/').pop()">
+                            <span t-if="not no_text" t-esc="lg.name.split('/').pop()" t-attf-class="#{_txt_class}"/>
+                            <span t-elif="codes" t-esc="lg.url_code.split('_').pop(0).upper()" t-attf-class="align-middle #{_txt_class}"/>
```

Comment:
There is more than the occurrences here to change, for instance there is a few ones in website too, you can grep `flag_image_src` which still does `lg[4]`, `active_lang[4]`
This is crashing, you can test with the following steps:
- Install website
- Install a second lang on website through the settings
- Enter edit mode on any page
- Click on the footer language switcher
- Change the display option to "Text and Flag"

It will reload and crash:
```
The error occurred while rendering the template website.layout and evaluating the following expression: <div id="website_cookies_bar" t-if="website.cookies_bar" class="s_popup o_snippet_invisible d-none o_no_save" data-name="Cookies Bar" data-vcss="001" data-invisible="1" data-oe-id="2199" data-oe-xpath="/data/xpath/div" data-oe-model="ir.ui.view" data-oe-field="arch"/>
```